wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java

```diff
@@ -163,59 +163,30 @@
   });
 
   private final Schema schema;
-  private final ImmutableMap<ProtoType, ClassName> nameToAbstractAdapterName;
   private final ImmutableMap<ProtoType, ClassName> nameToJavaName;
-  private final ImmutableMap<ProtoType, AdapterConstant> protoTypeToAdapterConstant;
+  private final Profile profile;
   private final boolean emitAndroid;
   private final boolean emitCompact;
 
-  private JavaGenerator(Schema schema,
-      ImmutableMap<ProtoType, ClassName> nameToAbstractAdapterName,
-      ImmutableMap<ProtoType, ClassName> nameToJavaName,
-      ImmutableMap<ProtoType, AdapterConstant> protoTypeToAdapterConstant,
+  private JavaGenerator(Schema schema, Map<ProtoType, ClassName> nameToJavaName, Profile profile,
       boolean emitAndroid, boolean emitCompact) {
     this.schema = schema;
-    this.nameToAbstractAdapterName = nameToAbstractAdapterName;
-    this.nameToJavaName = nameToJavaName;
-    this.protoTypeToAdapterConstant = protoTypeToAdapterConstant;
+    this.nameToJavaName = ImmutableMap.copyOf(nameToJavaName);
+    this.profile = profile;
     this.emitAndroid = emitAndroid;
     this.emitCompact = emitCompact;
   }
 
   public JavaGenerator withAndroid(boolean emitAndroid) {
```

Comment:
I’ll roll it into the profile in the follow up. I’m tempted to make this a full breaking change. Maybe call it Wire 3.0 ?